src/main/bootstrap/mod.rs

```diff
@@ -332,6 +330,18 @@ fn shuffled_bootstrap_peers(
     peers
 }
 
+fn update_peers(peers: &mut Vec<PeerInfo>, rx: Receiver<Vec<PeerInfo>>) {
+    let discovered_peers: Vec<PeerInfo> = {
+        let our_peers: HashSet<&PeerInfo> = HashSet::from_iter(peers.iter());
+        rx.try_iter()
+            .flatten()
+            .filter(|peer| !our_peers.contains(peer))
```

Comment:
since `rx` receives `Vec<PeerInfo>`, theoretically speaking it could receive duplicate items itself. So say `rx.try_iter()` yields `["192.168.1.100", "192.168.1.100"]` (don't mind me skipping ports), then this list will be `flattened()` and `.filter()` receives each address individually. Note that at the time `filter()` gets second address, it is not in `our_peers` yet, so it will be passed through filter and later collected to `discovered_peers`.
I'd say at the bery least make `discovered_peers` a `HashSet` and see if you can make `rx` into `Receiver<HashSet<..>>`.